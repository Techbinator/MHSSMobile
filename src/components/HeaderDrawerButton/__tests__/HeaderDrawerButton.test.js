import React from 'react';
import renderer from 'react-test-renderer';

import HeaderDrawerButton from '../index';

describe('Components : HeaderDrawerButton', () => {
  it('renders correctly', () => {
    const props = {
      navigation: { dispatch: jest.fn() },
    };
    const tree = renderer.create(<HeaderDrawerButton {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  /**
  it('toogles drawer when button is clicked', () => {
    const props = {
      navigation: { dispatch: jest.fn() },
    };
    const testRenderer = renderer.create(<HeaderDrawerButton {...props} />);
    const instance = testRenderer.root;
    const button = instance.findByType(Button);

    button.props.onPress();
    button.props.onPress();

    expect(props.navigation.dispatch.mock.calls).toEqual([
      ['toggleDrawer'],
      ['toggleDrawer'],
    ]);
  });
 */
});
