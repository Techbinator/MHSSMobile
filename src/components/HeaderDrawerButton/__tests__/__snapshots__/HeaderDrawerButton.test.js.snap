// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : HeaderDrawerButton renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "transparent",
      "borderBottomWidth": null,
      "borderColor": "#007aff",
      "borderLeftWidth": null,
      "borderRadius": 5,
      "borderRightWidth": null,
      "borderTopWidth": null,
      "borderWidth": undefined,
      "elevation": 0,
      "flexDirection": "row",
      "height": 45,
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingBottom": 6,
      "paddingTop": 6,
      "shadowColor": null,
      "shadowOffset": null,
      "shadowOpacity": null,
      "shadowRadius": null,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
</View>
`;
